### PR TITLE
feat: constify some basic functions

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

File: src/multihash.rs [hunk 1]
```diff
@@ -103,13 +103,17 @@ impl<const S: usize> Default for Multihash<S> {
...
             return Err(Error::InvalidSize(input_digest.len() as _));
         }
         let size = input_digest.len();
         let mut digest = [0; S];
-        digest[..size].copy_from_slice(input_digest);
+        let mut i = 0;
+        while i < size {
+            digest[i] = input_digest[i];
+            i += 1;
+        }
```
Comment (Stebalien) [inline src/multihash.rs:116]: `for i in 0..size`?

(and do you know if rust optimizes this well?)

**Review by Stebalien — APPROVED** [review]

File: src/multihash.rs (hunk 1 above)
Comment (dignifiedquire) [reply src/multihash.rs:116]: `for` is not stable in `const fn`

Comment (Stebalien) [reply src/multihash.rs:116]: Lol...

**Comment by Stebalien** [comment]
Looks like a new clippy lint. Mind telling clippy to shut up about that?

**Pull request merged**